src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/JsonSerializerContextTests.cs

```diff
@@ -21,6 +21,27 @@ public static void VariousNestingAndVisibilityLevelsAreSupported()
             Assert.NotNull(NestedPublicContext.NestedProtectedInternalClass.Default);
         }
 
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public static void ContextMetadataIsImmutable()
+        {
+            RemoteExecutor.Invoke(
+                static () =>
+                {
+                    JsonTypeInfo<Person> typeInfo = PersonJsonContext.Default.Person;
+
+                    Assert.Throws<InvalidOperationException>(() => typeInfo.CreateObject = null);
```

Comment:
Should these be written (partly?) with reflection to ensure that we don't miss any new properties in the future, similar to the `JsonSerializerOptions` copy ctor tests.

Reply:
I don't think it's necessary in this context. We have test coverage in System.Text.Json.Tests validating that once a metadata instance is frozen then all properties are non-user modifiable. 

This is just a smoke test validating that the instance is locked.